via/services/checkmate.py

```diff
@@ -5,13 +5,20 @@
 from via.exceptions import BadURL
 
 
-class ViaCheckmateClient(CheckmateClient):
-    def __init__(self, request):
-        super().__init__(
-            host=request.registry.settings["checkmate_url"],
-            api_key=request.registry.settings["checkmate_api_key"],
+class CheckmateService:
+    def __init__(self, checkmate_client, allow_all, blocked_for, ignore_reasons):
+        self._checkmate_client = checkmate_client
+        self._allow_all = allow_all
+        self._blocked_for = blocked_for
+        self._ignore_reasons = ignore_reasons
+
+    def check_url(self, url):
```

Comment:
```suggestion
    def check_url(self, url): -> Optional[BlockResponse]
```

I think this is handy to add as the block response isn't just a boolean or something, but has structure.

A doc-string here would be good. I'd be in favour of re-introducing doc-strings as a required part of our linting. I don't think missing them out is an exceptional case at the moment.

Reply:
:+1: https://github.com/hypothesis/via/pull/1004/commits/cf2604d2bfd54dd4e46625a7eff2fd87e9177ad9 It's going to be so much nicer when we can write `-> BlockResponse | None` instead of using `Optional` (Python 3.10)

Reply:
Added a docstring too: https://github.com/hypothesis/via/pull/1004/commits/dc41000eac2c40de9f8815c2278d42704c842936